x-pack/platform/plugins/shared/streams/server/lib/content/saved_object.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 } from 'uuid';
+import {
+  ContentPackIncludedObjects,
+  ContentPackSavedObject,
+  INDEX_PLACEHOLDER,
+  findIndexPatterns,
+  isIncludeAll,
+  replaceIndexPatterns,
+} from '@kbn/content-packs-schema';
+import { compact, uniqBy } from 'lodash';
+
+export function prepareForExport({
+  savedObjects,
+  source,
+  replacedPatterns = [],
+}: {
+  savedObjects: ContentPackSavedObject[];
+  source: string;
+  replacedPatterns?: string[];
+}) {
+  return savedObjects.map((object) => {
+    if (object.type === 'dashboard' || object.type === 'index-pattern') {
+      const patterns = findIndexPatterns(object);
+      const replacements = {
+        ...replacedPatterns.reduce((acc, pattern) => {
+          acc[pattern] = INDEX_PLACEHOLDER;
+          return acc;
+        }, {} as Record<string, string>),
+
+        ...patterns
+          .filter((pattern) => pattern.startsWith(source))
```

Comment:
I'm not sure about checking for prefix here...

At least for wired streams, the "correct" index pattern is `{stream.name},{stream.name}.*` - so either the stream name exact match or stream name, then a dot, then something else. For classic streams of course we can only guess. I don't know where this leaves us - maybe `startsWith` is close enough that it will work in practice. Thoughts, @tommyers-elastic @LucaWintergerst ?

Reply:
The pattern replacement that has both an implicit and explicit logic:
- any detected pattern that has the source stream as a prefix (classic or wired) will be automatically replaced (only the prefix part)
- other patterns that do not satisfy the previous condition are surfaced in the UI at export time. user can select which one needs to be replaced. The selected pattern will currently be replaced by the target stream pattern, but this logic can be extended to set custom patterns

If a dashboard defines `{stream.name},{stream.name}.*` these will be correctly replaced, for unwired streams we ask for user input. @flash1293 is your concern about the detecton itself (which should be ```pattern === source || pattern.startsWith(`${source}.`)```) ?